pkg/cmd/events/tail/tail.go

```diff
@@ -0,0 +1,163 @@
+	"github.com/algolia/cli/pkg/validators"
+
+	_insights "github.com/algolia/algoliasearch-client-go/v3/algolia/insights"
+	region "github.com/algolia/algoliasearch-client-go/v3/algolia/region"
+	"github.com/algolia/cli/api/insights"
+)
+
+const (
+	// DefaultRegion is the default region to use.
+	DefaultRegion = "us"
+
+	// Interval is the interval between each request to fetch events.
+	Interval = 3 * time.Second
+)
+
+// TailOptions contains all the options for the `events tail` command.
+type TailOptions struct {
+	Config config.IConfig
+	IO     *iostreams.IOStreams
+
+	SearchClient func() (*search.Client, error)
+
+	Region string
+
+	PrintFlags *cmdutil.PrintFlags
+}
+
+// NewTailCmd returns a new command for tailing events.
+func NewTailCmd(f *cmdutil.Factory, runF func(*TailOptions) error) *cobra.Command {
+	opts := &TailOptions{
+		IO:           f.IOStreams,
+		Config:       f.Config,
+		SearchClient: f.SearchClient,
+		PrintFlags:   cmdutil.NewPrintFlags(),
+	}
+	cmd := &cobra.Command{
+		Use:   "tail",
+		Args:  validators.NoArgs(),
+		Short: "Tail events",
+		Long: heredoc.Doc(`
+			Tail events from your Algolia application.
+
+			By default, this command will tail events for the United States region.
+			If your Analytics data is stored in a different region than the United States (e.g. Germany/Europe), you can specify the region using the --region (-r) flag.
+		`),
+		Example: heredoc.Doc(`
+			# Tail events
+			$ algolia events tail
+
+			# Tail events for a specific region matching the Analytics region of your application
+			$ algolia events tail -r de
+
+			# Tail events and output them as JSON
+			$ algolia events tail --output json
+		`),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if runF != nil {
+				return runF(opts)
+			}
+
+			return runTailCmd(opts)
+		},
+	}
+
+	cmd.Flags().StringVarP(&opts.Region, "region", "r", DefaultRegion, "Region where your analytics data is stored and processed.")
+	_ = cmd.RegisterFlagCompletionFunc("region", cmdutil.StringCompletionFunc(map[string]string{
+		"us": "United States",
+		"de": "Germany (Europe)",
+	}))
+
+	opts.PrintFlags.AddFlags(cmd)
+
+	return cmd
+}
+
+func runTailCmd(opts *TailOptions) error {
+	appID, err := opts.Config.Profile().GetApplicationID()
+	if err != nil {
+		return err
+	}
+	apiKey, err := opts.Config.Profile().GetAdminAPIKey()
+	if err != nil {
+		return err
+	}
+
+	// We don't use the base insights client because it doesn't support fetching events.
+	config := _insights.Configuration{
+		AppID:  appID,
+		APIKey: apiKey,
+		Region: region.Region(opts.Region),
+	}
+	insightsClient := insights.NewClientWithConfig(config)
+
+	var p printers.Printer
+	if opts.PrintFlags.OutputFlagSpecified() && opts.PrintFlags.OutputFormat != nil {
+		p, err = opts.PrintFlags.ToPrinter()
+		if err != nil {
+			return err
+		}
+	} else if opts.IO.IsStdoutTTY() {
+		fmt.Fprint(opts.IO.Out, "\nWaiting for events... Press Ctrl+C to stop.\n")
```

Comment:
Maybe it's just me being a "keke tuning 😆" but maybe it'd be nice to have an animated indicator using `https://github.com/briandowns/spinner` here, wdyt?

Reply:
I tried it (we are already using the spinner package btw), but it's breaking the output when we are starting to receive the events. Will try again to see if I can fix the issue.

Reply:
Welp, I tried, but it seems we cannot have on the same shell an animation on one line and other line showing up at the same time. So it will be for another time I guess.